drivers/wifi/nrf700x/src/wifi_mgmt.c

```diff
@@ -739,6 +739,15 @@ int nrf_wifi_mode(const struct device *dev,
 
 	} else {
 		mode->mode = def_dev_ctx->vif_ctx[vif_ctx_zep->vif_idx]->mode;
+
+		/**
+		 * This is a work-around to maintain umac context for mode same as much
+		 * as possible. Only provide major modes to the user
```

Comment:
(Generic suggestion) Maybe we can call them `primary modes` instead of `major modes`